database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -75,45 +77,48 @@ public void onChildAdded(DataSnapshot snapshot, String previousChildKey) {
         notifyChangedListeners(OnChangedListener.EventType.ADDED, index);
     }
 
+    @Override
     public void onChildChanged(DataSnapshot snapshot, String previousChildKey) {
         int index = getIndexForKey(snapshot.getKey());
         mSnapshots.set(index, snapshot);
         notifyChangedListeners(OnChangedListener.EventType.CHANGED, index);
     }
 
+    @Override
     public void onChildRemoved(DataSnapshot snapshot) {
         int index = getIndexForKey(snapshot.getKey());
         mSnapshots.remove(index);
         notifyChangedListeners(OnChangedListener.EventType.REMOVED, index);
     }
 
+    @Override
     public void onChildMoved(DataSnapshot snapshot, String previousChildKey) {
         int oldIndex = getIndexForKey(snapshot.getKey());
         mSnapshots.remove(oldIndex);
-        int newIndex = previousChildKey == null ? 0 : (getIndexForKey(previousChildKey) + 1);
+        int newIndex = previousChildKey == null ? 0 : getIndexForKey(previousChildKey) + 1;
```

Comment:
Please put back the parentheses around `(getIndexForKey(previousChildKey) + 1)`. The behavior is the same, but it makes the intent more explicit.